reactioncommerce/reaction
# Plugin: Prometheus Metrics

Resolves [reaction-feature-requests/60](https://github.com/reactioncommerce/reaction-feature-requests/issues/60)  
Impact: **minor**  
Type: **feature**

## Issue

Currently reaction is not instrumented for application performance monitoring.

## Solution

- Add a new included plugin which Incorporates the npm package `epimetheus`
- This will expose prometheus metrics about the node process
  - It's disabled by default
  - Can be enabled in the Catalyst UI
  - The HTTP path it uses defaults to `/metrics` but is configurable in the catalyst UI

## Breaking changes

Potential for http routing conflicts on `/metrics`.

## Testing

- Fire up reaction core with this branch
- Navigate to the catalyst (operator UI)
- Click on "Metrics Prometheus"
- Check "enabled"
- Click "Save Changes"
- Restart reaction
  - (Although technically we could probably make this changeable without a restart, it would be a lot more work)
- `curl localhost:3000/metrics` should spit out some plain text prometheus metrics
- Go back and disable this plugin
- restart reaction
- `curl localhost:3000/metrics` should give you a 404 status code and the "Not Found" HTML page
